storage/transaction.go

```diff
@@ -47,6 +47,26 @@ func (s *Storage) unsafeGetTransactionWithBatchDetails(hash common.Hash) (
 	return result, nil
 }
 
+func (s *TransactionStorage) GetTransactionsByCommitmentID(id models.CommitmentID) (models.GenericTransactionArray, error) {
+	batchedTxs := make([]stored.BatchedTx, 0, 32)
+
+	query := bh.Where("CommitmentID").Eq(id).Index("CommitmentID")
+
+	err := s.database.Badger.Find(&batchedTxs, query)
+	if err != nil {
+		return nil, err
+	}
+
+	var txs models.GenericTransactionArray
+
+	txs = models.MakeGenericArray()
+	for i := range batchedTxs {
+		txs = txs.AppendOne(batchedTxs[i].ToGenericTransaction())
+	}
+
+	return txs, nil
```

Comment:
What do you think about below change? It's simpler and we create slice with target capacity.
```suggestion
	txs := make(models.GenericArray, 0, len(batchedTxs))
	for i := range batchedTxs {
		txs = append(txs, batchedTxs[i].ToGenericTransaction())
	}
	return txs, nil
```

Reply:
Huh, I thought I tried this and got a type error, but trying it again now this works, nice catch!